frontend/src/components/eventsModal.js

```diff
@@ -0,0 +1,88 @@
+import { Component } from 'react'
+import Router from 'next/router'
+import {
+  Alert,
+  Button,
+  Form,
+  FormGroup,
+  Label,
+  Input,
+  Modal,
+  ModalHeader,
+  ModalBody,
+  ModalFooter
+} from 'reactstrap'
+
+class EventsModal extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      alert: ''
+    }
+  }
+
+  handleKeyPress = async event => {
+    if (event.key === 'Enter') {
+      this.handleSubmit()
+    }
+  }
+
+  handleSubmit = async () => {
+    const success = await this.props.onSubmit()
+    if (success) {
+      Router.push({ pathname: this.props.pathname })
+    } else {
+      this.setState({
+        alert: this.props.alert
+      })
+    }
+  }
+
+  closeModal = () => {
+    // remove err message upon close
+    if (this.state.alert) {
+      this.setState({
+        alert: ''
+      })
+    }
+    this.props.toggle()
+  }
+
+  render() {
+    const alert = <Alert color="danger">{this.state.alert}</Alert>
+    return (
+      <>
```

Comment:
The fragment is unnecessary here. The main use case is when you have multiple components at the top most level (like `Head`, `Nav`, etc). Here the only top level component is `Modal` so the fragment is unnecessary.